lib/rules/no-restricted-imports.js

```diff
@@ -235,9 +241,10 @@ module.exports = {
 
         // relative paths are supported for this rule
         const restrictedPatternGroups = restrictedPatterns.map(
-            ({ group, message, caseSensitive, importNames, importNamePattern, allowImportNames, allowImportNamePattern }) => (
+            ({ group, regex, message, caseSensitive, importNames, importNamePattern, allowImportNames, allowImportNamePattern }) => (
                 {
-                    matcher: ignore({ allowRelativePaths: true, ignorecase: !caseSensitive }).add(group),
+                    ...(group ? { matcher: ignore({ allowRelativePaths: true, ignorecase: !caseSensitive }).add(group) } : {}),
+                    ...(regex ? { regexMatcher: new RegExp(regex, caseSensitive ? "u" : "iu") } : {}),
```

Comment:
```suggestion
                    ...(typeof regex === "string" ? { regexMatcher: new RegExp(regex, caseSensitive ? "u" : "iu") } : {}),
```

To avoid crashing when `regex` is an empty string.